helix-lsp/src/lib.rs

```diff
@@ -58,6 +58,38 @@ pub mod util {
     use super::*;
     use helix_core::{Range, Rope, Transaction};
 
+    /// Converts a diagnostic in the document to [`lsp::Diagnostic`].
+    ///
+    /// Panics when [`pos_to_lsp_pos`] would for an invalid range on the diagnostic.
+    pub fn diagnostic_to_lsp_diagnostic(
+        doc: &Rope,
+        diag: &helix_core::diagnostic::Diagnostic,
+        offset_encoding: OffsetEncoding,
+    ) -> lsp::Diagnostic {
+        use helix_core::diagnostic::Severity::*;
+
+        let severity = diag.severity.map(|s| match s {
+            Hint => lsp::DiagnosticSeverity::HINT,
+            Info => lsp::DiagnosticSeverity::INFORMATION,
+            Warning => lsp::DiagnosticSeverity::WARNING,
+            Error => lsp::DiagnosticSeverity::ERROR,
+        });
+
+        // TODO: add support for Diagnostic.data
+        lsp::Diagnostic::new(
+            lsp::Range::new(
+                pos_to_lsp_pos(doc, diag.range.start, offset_encoding),
+                pos_to_lsp_pos(doc, diag.range.end, offset_encoding),
+            ),
```

Comment:
Use `range_to_lsp_range`: https://github.com/helix-editor/helix/blob/7c6cd8957e4f3d4399664fc8e37854f4b3261b3e/helix-lsp/src/lib.rs#L151-L161=

Reply:
That takes a `helix_core::selection::Range` whereas the diagnostics have their own `helix_core::diagnostic::Range`.
So either this was an option, or you'd need to construct the regular Range first.